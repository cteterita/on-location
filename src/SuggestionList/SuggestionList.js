import React from 'react';
import PropTypes from 'prop-types';

import Suggestion from '../Suggestion/Suggestion';

function SuggestionList(props) {
  const { suggestions } = props;
  return (
    <ul>
      {suggestions.map((s) => {
        const { lat, lon } = s;
        return (
          <Suggestion
            displayName={s.display_name}
            lat={lat}
            lon={lon}
            key={s.place_id}
          />
        );
      })}
    </ul>
  );
}

SuggestionList.propTypes = {
  suggestions: PropTypes.arrayOf(
    PropTypes.shape({
      display_name: PropTypes.string,
      place_id: PropTypes.int,
    }),
  ),
};

SuggestionList.defaultProps = {
  suggestions: [],
};

export default SuggestionList;
