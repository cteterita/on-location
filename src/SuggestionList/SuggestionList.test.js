import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { render, screen } from '@testing-library/react';

import SuggestionList from './SuggestionList';

const exampleSuggestions = [
  {
    display_name: 'example 1',
    place_id: 1,
    lat: 1,
    lon: 1,
  },
  {
    display_name: 'example 2',
    place_id: 2,
    lat: 1,
    lon: 1,
  },
];

test('renders without errors', () => {
  render(<BrowserRouter><SuggestionList suggestions={exampleSuggestions} /></BrowserRouter>);
  expect(screen.getByRole('list')).toMatchSnapshot();
});
