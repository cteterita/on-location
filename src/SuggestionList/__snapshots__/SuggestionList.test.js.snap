// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without errors 1`] = `
<ul>
  <li>
    example 1
  </li>
  <li>
    example 2
  </li>
</ul>
`;
